src/instance.ts

```diff
@@ -166,6 +232,159 @@ export class PtzOpticsInstance extends InstanceBase<PtzOpticsConfig> {
+
+					if (result?.data) {
+						variableDefinitions = [
+							{
+								name: 'HTTP Device Name',
+								variableId: 'HTTPdevname',
+							},
+							{
+								name: 'HTTP Device Version',
+								variableId: 'HTTPdevVersion',
+							},
+							{
+								name: 'HTTP Serial Number',
+								variableId: 'HTTPserialNum',
+							},
+							{
+								name: 'HTTP Device Model',
+								variableId: 'HTTPdeviceModel',
+							},
+						]
+						result.data.device_model = result.data.device_model.trim()
+						model = result.data.device_model
+						devVersion = result.data.versioninfo
+						variableValues = {
+							HTTPdevname: result.data.devname,
+							HTTPdevVersion: devVersion,
+							HTTPserialNum: result.data.serial_num,
+							HTTPdeviceModel: result.data.device_model,
+						}
+					}
+				} catch (err) {
+					this.log('error', `device config fetch failed: ${err}`)
+				}
+				// check firmware version
+				try {
+					this.#digestClient2 = new DigestClient('', '')
+					const result = (await this.#digestClient2.fetch(`https://firmware.ptzoptics.com/${model}/RVU.json`, {
+						method: 'GET',
+						headers: {
+							'User-Agent': 'Mozilla/5.0',
+							Accept: 'application/json',
+						},
+					})) as Response
+					if (!result.ok) {
+						throw new Error(`HTTP ${result.status}`)
+					}
+					const text = await result.text()
+					let parsed: any
+					try {
+						parsed = JSON.parse(text)
+					} catch {
+						throw new Error(`Unexpected content type, could not parse JSON.`)
+					}
+					devVersion = devVersion.split(' v')[1] || ''
+					variableDefinitions.push({ name: 'HTTP Device Updateable', variableId: 'HTTPdevUpdateable' })
+					if (parsed.data.soc_version !== devVersion) {
+						this.log(
+							'info',
+							`Firmware update from ${devVersion} to ${parsed.data.soc_version} available. Changelog: https://firmware.ptzoptics.com/${model}/${parsed.data.log_name}`,
+						)
+						variableValues = { ...variableValues, HTTPdevUpdateable: parsed.data.soc_version }
+					} else {
+						variableValues = { ...variableValues, HTTPdevUpdateable: 0 }
+					}
+				} catch (err) {
+					this.log('error', `PTZ Firmware fetch failed: ${err}`)
+				}
+			}
+			// start tally-polling, if interval > 0
+			const interval = Number(config.HTTPpollInterval)
+			if (interval > 0 && username && password) {
+				let isRequestInProgress = false
+				let firstRun = true
+				this.tallyPollTimer = setInterval(() => {
+					void (async () => {
+						if (isRequestInProgress) return // skip if already/still running
+						// Get tally status
+						try {
+							isRequestInProgress = true
+							const result = await this.sendHTTPCommand<{ data: any }>('/cgi-bin/param.cgi?get_tally_status', 'GET')
+							if (result?.data) {
+								if (firstRun) {
+									for (const key of Object.keys(result.data)) {
+										const varId = `HTTP${key}`
```

Comment:
Oh, hmm, you're directly tying the variable IDs to what's in this CGI API.  I think I would prefer you don't do that, and that you specifically select out the keys you're using and manually map them to the variable IDs.  Implicit dependencies (especially coming from an API that just returns more or less JSON) make code harder to understand.